src/mlem.rs

```diff
@@ -216,6 +220,25 @@ impl Image {
     }
 }
 
+/// Define the scatter prediction histogram (incomplete)
+pub fn scatter_prediction(lors: &[LOR]) -> Option<Scattergram<JustZ>>
+{
+    // This function is not really what I want but it gets me started.
+    // Should probably allow type of Lorogram as parameter. JustZ for now
+    let mut scatters = Scattergram::new(JustZ::new(2000.0, 20));
+
+    // Just call every 10th LOR scatter for now.
```

Comment:
I thought about that but here I don't have access to the energy variables because it's a `LOR` and not an `HDF5Lor`. In any case, this function is a dummy test, not really much use for the final problem.

Reply:
`HDF5Lor` maps directly to how they are stored on disk. `LOR` is how they are represented in memory for the purposes of MLEM, and there are ready conversions between the two.

Did you manage to download a data file which gives something nontrivial in `show_lororgam`?

As soon as it compiles, we'll want to start looking at whether it makes any sense at all, and for that you need real data, which you will read in as `HDF5Lor`s.

`io::hdf5::read_lors` already returns `LOR`s, so `bin/mlem.rs` doesn't see any `LOR`s. Clearly that will have to change.

Reply:
> HDF5Lor maps directly to how they are stored on disk. LOR is how they are represented in memory for the purposes of MLEM, and there are ready conversions between the two.

I noticed, in the long term it certainly seems correct as is since the scatter PDF would be generated separately, right? For the current tests I'd need access somehow. Maybe I can just do an additional loop over the data for now, not so efficient but since it'll change soon anyway.

> Did you manage to download a data file which gives something nontrivial in show_lororgam?

I have a file, the huge one that I started copying timed out but I found a small example. I'll start testing later on.

Reply:
Yes, in the very long term it's probably correct. For the time being we'll want to be generating the scattergrams on the fly. I'd go with a second loop to start with, and only do something smarter if the waste of time starts getting annoying.

>>Did you manage to download a data file which gives something nontrivial in `show_lororgam`?

> I have a file, the huge one that I started copying timed out but I found a small example. I'll start testing later on.

Check that `show_lorogram` gives something other than `1.0`s, sooner rather than later.

Reply:
> Check that show_lorogram gives something other than 1.0s, sooner rather than later.

I checked yesterday, some values other than `1.0` confirmed. I have to work out a nix thing at the moment, the commands have disappeared after a MAC update...

Reply:
Fixed and confirmed that I have some `1.1`s in my example file. I'll implement the loop and make the updates you suggested and then maybe some peras would be useful to try and get it in a more generally useful state.

Reply:
`1.1`s suggests that there's not much scatter in there at all. This is what I get:

1.0
3.0
3.0
2.9
2.9
2.7
2.7
2.3
2.3
1.9
1.9
2.0
2.0
2.4
2.4
2.8
2.8
3.0
3.0
3.1
3.1
1.0